core/src/main/java/io/kestra/core/schedulers/AbstractScheduler.java

```diff
@@ -163,17 +216,31 @@ public void run() {
                         workerTriggerResult.getTriggerContext()
                     );
                     this.handleEvaluatePollingTriggerResult(triggerExecution);
-                }
-                else {
+                } else {
                     // previously, if no interval the trigger was executed immediately. I think only the Schedule trigger has no interval
                     // now that all triggers are sent to the worker, we need to do this to avoid issues with backfills
                     // as the same trigger will be evaluated multiple-time which is not supported by the 'addToRunning' method
                     // if the trigger didn't trigger an execution, we will need to clear the evaluation lock
                     // TODO now that Schedule is executed by the Scheduler this could be relaxed
                     if (workerTriggerResult.getTrigger() instanceof PollingTriggerInterface &&
                         ((PollingTriggerInterface) workerTriggerResult.getTrigger()).getInterval() != null) {
-                        var triggerNotRunning = Trigger.of(workerTriggerResult.getTriggerContext());
+
+                        ZonedDateTime nextExecutionDate = now().plus(((PollingTriggerInterface) workerTriggerResult.getTrigger()).getInterval());
```

Comment:
You can use trigger.nextEvaluationDate if you update it's default implementation for now().plus(getIntervall) which seems more accurate than now